src/events.ts

```diff
@@ -21,6 +21,12 @@ type DeviceDisconnectEventPayload = {
 
 // Devices
 export type DeviceConnectedEvent = CommonDeviceEvent<"device.connected">
+
+export type DeviceConvertedToUnmanagedEvent =
+  CommonDeviceEvent<"device.converted_to_unmanaged">
+export type UnmanagedDeviceConvertedToManagedEvent =
+  CommonDeviceEvent<"device.unmanaged.converted_to_managed">
```

Comment:
It seems like the event should be redundant for both namespaces in each direction, e.g. `device.converted_from_unmanaged` and `device.unmanaged.converted_from_managed` should be present in addition to these. This is consistent with our approach of "redundant endpoints within a relevant user story namespace" e.g. `locks.list` is redundant with `devices.list`, but it's helpful to have the namespace contained.

To be clear the proposed change is an improvement and solves the customer issue.

Reply:
I am in favor of this, but we should make the update for access codes as well. You could go further and say we should have redundant events for all device subtypes: `lock.connected` / `thermostat.connected` / `lock.converted_from_unmanaged` etc.

Reply:
we could also have a rule related to lifecycles, e.g. "the original resource type has preference for conversion", in which case the current events are correct.

Reply:
Yes, I agree about the "more root" rule.

FWIW I'm almost certain we discussed what events to use for this conversion case when we did unmanaged access codes. The conversion case is unique because the resource is actually moved.

~~I believe the core reason we have it this way (which may have been noted at the decision point)  is because the `device_id` in the event will be usable in the new namespace.~~

~~If, as now suggested, we emit `device.converted_from_unmanaged { device_id }` then event consumers trying to `GET /device/get?device_id=device_id` would see a 404. So with that observation, I don't think I support the other events now.~~

Hmm, so I just wrote out the behavior that made sense to me, doubled checked the events, and saw we actually do the opposite, e.g., we send `device.converted_to_unmanaged { device_id }` BUT `GET /device/get?device_id=device_id => 404` and `device.unmanaged.converted_to_managed { device_id }` BUT `GET /device/unmanaged/get?device_id=device_id => 404`.

So options:

1. Keep what we have now.
2. Add symmetric events. Agree that the behavior where some events contain a primary id that does not exist in the namespace is ok (it's the behavior we might have if we ever implemented a `device.deleted` event).
3. Reconsider if we should not have events that contain a primary id that does not exist in the event namespace.

Reply:
Going to think about this more, it seems like (2) might be the way to go and resource conversion is symmetric generally, but i think its worth documenting and discussing this in our api design rfc doc, making a note to add something there